### ISSUE TITLE
Using locale in Power BI Desktop

### ISSUE THREAD

**Issue by MitjaBezensek** [Microsoft/PowerBI-visuals#292]
I'm trying to localise the display of numbers in our custom visual. I was able to do it for the online version by using `options.host.locale` as exaplained in the [simpleBarChart tutorial](https://github.com/Microsoft/PowerBI-visuals-sampleBarChart/blob/master/Tutorial/Locale.md). 

I have set the Regional settings inside of Power BI Desktop to German:
![locale-pbi](https://user-images.githubusercontent.com/2523721/34557573-010c3714-f13b-11e7-910d-d883f0cac26c.png)

And also my Windows Regional settings are German:
![locale-cp](https://user-images.githubusercontent.com/2523721/34557588-14a91c6a-f13b-11e7-9cf4-e65b0701a947.png)

However, if I now use the custom visual in the Desktop version it does not take my locale into consideration. As you can see it is still using dot as a decimal separator, when it should use a comma:
![image](https://user-images.githubusercontent.com/2523721/34557639-3c7081ca-f13b-11e7-8260-ecdbc42a7a7d.png)

I even packaged the BarChart example and in the desktop version it also shows the language as English:
![image](https://user-images.githubusercontent.com/2523721/34557669-5531b60c-f13b-11e7-97e6-c6d53fe8cc3a.png)

While it correctly works online:
![image](https://user-images.githubusercontent.com/2523721/34557698-6f7eca04-f13b-11e7-86fe-df7d71f2165b.png)

What might be causing this? Is there some other setting that I'm not aware of? Or maybe different locales are not currently supported in the desktop version?

**Comment by uve** [comment]
Hello @MitjaBezensek, what visual are you using?

**Comment by MitjaBezensek** [comment]
@uve we are developing our custom visual, but you can test this with the [simple bar chart](https://github.com/Microsoft/PowerBI-visuals-sampleBarChart/) as well. You can see the screenshots of the simple bar chart example above.

Another option is to create a simple visual that just displays `options.host.locale`. You'll see that it's always the same in the desktop version. It doesn't display the selected regional or windows regional settings.

**Comment by uve** [comment]
@MitjaBezensek, Desktop app doesn't use system locale, you have to download localized version of Desktop to use any other language. https://www.microsoft.com/en-us/download/details.aspx?id=45331
Custom visual gets locale string from Desktop App. So if you want to use different formatting in your custom visual, I suggest to use this util in your code:
https://github.com/Microsoft/powerbi-visuals-utils-formattingutils/blob/master/docs/api/value-formatter.md#the-culture-selector

**Comment by MitjaBezensek** [comment]
@uve So if I understand this correctly the visuals that come with Power BI use the system locale and the custom visuals use the locale from the Desktop App?

1. Are there any plans to support the system locale for custom visuals? I think our customers would prefer an unified experience regarding this.
2. What exactly does the Regional setting in Power BI do?

**Comment by uve** [comment]
@MitjaBezensek, no, all visuals use locale from Desktop App only.

**Comment by MitjaBezensek** [comment]
At least from my tests this doesn't seem to be the case. 

Please check the following gif where I open report with Slovenian language settings (you can see the comma used as decimal separator). After that I change the language settings to English and reopen the report. Now dot is used as decimal separator.

You can open it in new tab to see the full resolution:
![visuals](https://user-images.githubusercontent.com/2523721/34761498-63967fc2-f5e4-11e7-8c24-76f9d852eb61.gif)

**Comment by uve** [comment]
Are you checking it on BarChart, is that correct? it's core visual and it doesn't use new API yet.
At the same time, new API provide options.host.locale to CustomVisual based on Desktop language only.

**Comment by MitjaBezensek** [comment]
I used the clustered column chart:
![image](https://user-images.githubusercontent.com/2523721/34830639-ca94145e-f6e4-11e7-85f7-ed45d149e194.png)

**Comment by uve** [comment]
yes, it's core visual, that's why it can use some features which are not available for Custom Visuals.

**Comment by MitjaBezensek** [comment]
Ok, now it's clear. 

That's why I asked if this behavior will be unified in the future? Because right now you can have two visuals side by side and they use different decimal separators, since one takes them from the system settings and the other one (custom visual) from the Desktop app settings.

**Comment by capono** [comment]
We have a similar issue, while using the AdvancedFilter in the Power BI Desktop. We are not able to filter comma number since the format does not match with the one Power BI using to filter.

So having the language set to german in Windows which use ',' for comma, and having a report set to English that use '.' for comma it is not possible within the Custom VIsual to use the a correct filter on comma numbers, since the format can not be determined correctly.

**Comment by zBritva** [comment]
@capono why do you pass numbers into AdvancedFilter as text?

**Comment by capono** [comment]
> @capono why do you pass numbers into AdvancedFilter as text?

We do this because otherwise it does not filter comma numbers that use ',' when the region settings are in German. 

It seems that the filters depend on the region format settings in Windows .

![image](https://user-images.githubusercontent.com/24607802/67679251-f737ab80-f988-11e9-83b1-d71decccd17e.png)

**Comment by zBritva** [comment]
So, you have numbers in data set, and they are texts, right?

**Comment by capono** [comment]
The data type is "Decimal Number".

If you use English as the region format setting on Windows, it looks like this:

![image](https://user-images.githubusercontent.com/24607802/67694810-450ddd00-f9a4-11e9-9d3a-02b28fc3a5d6.png)

When using German as region format settings under Windows, it looks like this : 

![image](https://user-images.githubusercontent.com/24607802/67694396-8ce03480-f9a3-11e9-9220-e8e5964dbb06.png). 

The filter pane updates the format correctly, but that does not happen in the "Custom Visuals" even the "Core Visuals" work correctly.

**Comment by zBritva** [comment]
@capono could you share the AdvancedFilter sample also? (ex. output of JSON.stringify(filter, null, " "))

**Comment by TimoRiikonen** [comment]
Hello,
Can you test if these instructions work:
https://community.powerbi.com/t5/Community-Blog/Localization-in-Power-BI/ba-p/952439

So the idea is to define German language on Desktop, because your data is in German. Convert values as numbers and I would expect that then you would not have issues any more and you could use any language to display the end result.
